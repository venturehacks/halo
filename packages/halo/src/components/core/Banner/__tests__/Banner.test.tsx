import { mount, render, shallow } from 'enzyme';
import React from 'react';

import { Banner } from '../index';

describe('Banner', () => {
  test('smoke', () => {
    const component = shallow(<Banner>PHP</Banner>);
    expect(component.text()).toEqual('PHP');
  });

  describe('dismiss', () => {
    test('hides on click', () => {
      const dismiss = jest.fn();
      const wrapper = mount(<Banner onDismiss={dismiss} offerDismiss />);

      let component = wrapper.find(`[data-test="Banner"]`);
      expect(component.hasClass('dismissed')).toBe(false);

      wrapper.find(`[data-test="Banner-closeButton"]`).last().simulate('click');

      component = wrapper.find(`[data-test="Banner"]`);
      expect(component.hasClass('dismissed')).toBe(true);
    });

    test('callback executes', () => {
      const dismiss = jest.fn();
      const wrapper = mount(<Banner onDismiss={dismiss} offerDismiss />);

      expect(dismiss).toBeCalledTimes(0);

      wrapper.find(`[data-test="Banner-closeButton"]`).last().simulate('click');

      expect(dismiss).toBeCalledTimes(1);
    });
  });

  describe('snapshot', () => {
    test('default', () => {
      const component = render(<Banner />);
      expect(component).toMatchSnapshot();
    });

    test('constrain', () => {
      const component = render(<Banner constrain />);
      expect(component).toMatchSnapshot();
    });

    test('offer dismiss', () => {
      const component = render(<Banner offerDismiss />);
      expect(component).toMatchSnapshot();
    });

    describe('variant', () => {
      test('success', () => {
        const component = render(<Banner variant="success" />);
        expect(component).toMatchSnapshot();
      });

      test('warning', () => {
        const component = render(<Banner variant="warning" />);
        expect(component).toMatchSnapshot();
      });

      test('error', () => {
        const component = render(<Banner variant="error" />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
