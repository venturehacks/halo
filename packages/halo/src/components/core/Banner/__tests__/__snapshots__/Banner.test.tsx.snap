// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner snapshot constrain 1`] = `
<div
  class="text-md font-medium rounded relative p-4 mb-2 border-solid border-0 border-t-4 w-full bg-blue-100 shadow-sm flex flex-row justify-start items-center overflow-hidden max-w-screen-desktop border-slate-700"
  data-test="Banner"
  role="alert"
>
  <aside
    class="mr-2 ml-1"
  />
  <div
    class="flex-1"
  />
</div>
`;

exports[`Banner snapshot default 1`] = `
<div
  class="text-md font-medium rounded relative p-4 mb-2 border-solid border-0 border-t-4 w-full bg-blue-100 shadow-sm flex flex-row justify-start items-center overflow-hidden max-w-screen-desktop border-slate-700"
  data-test="Banner"
  role="alert"
>
  <aside
    class="mr-2 ml-1"
  />
  <div
    class="flex-1"
  />
</div>
`;

exports[`Banner snapshot offer dismiss 1`] = `
<div
  class="text-md font-medium rounded relative p-4 mb-2 border-solid border-0 border-t-4 w-full bg-blue-100 shadow-sm flex flex-row justify-start items-center overflow-hidden max-w-screen-desktop border-slate-700"
  data-test="Banner"
  role="alert"
>
  <aside
    class="mr-2 ml-1"
  />
  <div
    class="flex-1"
  />
  <button
    class="text-sm pl-2 bg-transparent border-0 cursor-pointer text-dark-link font-normal"
    data-test="Banner-closeButton"
  >
    Dismiss
  </button>
</div>
`;

exports[`Banner snapshot variant error 1`] = `
<div
  class="text-md font-medium rounded relative p-4 mb-2 border-solid border-0 border-t-4 w-full bg-blue-100 shadow-sm flex flex-row justify-start items-center overflow-hidden max-w-screen-desktop border-red-600"
  data-test="Banner"
  role="alert"
>
  <aside
    class="mr-2 ml-1"
  >
    <svg
      class="block text-red-700 w-6"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m13.414 12 5.293-5.293a.999.999 0 1 0-1.414-1.414L12 10.586 6.707 5.293a.999.999 0 1 0-1.414 1.414L10.586 12l-5.293 5.293a.999.999 0 1 0 1.414 1.414L12 13.414l5.293 5.293a.997.997 0 0 0 1.414 0 .999.999 0 0 0 0-1.414L13.414 12Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </aside>
  <div
    class="flex-1"
  />
</div>
`;

exports[`Banner snapshot variant success 1`] = `
<div
  class="text-md font-medium rounded relative p-4 mb-2 border-solid border-0 border-t-4 w-full bg-blue-100 shadow-sm flex flex-row justify-start items-center overflow-hidden max-w-screen-desktop border-green-500"
  data-test="Banner"
  role="alert"
>
  <aside
    class="mr-2 ml-1"
  />
  <div
    class="flex-1"
  />
</div>
`;

exports[`Banner snapshot variant warning 1`] = `
<div
  class="text-md font-medium rounded relative p-4 mb-2 border-solid border-0 border-t-4 w-full bg-blue-100 shadow-sm flex flex-row justify-start items-center overflow-hidden max-w-screen-desktop border-orange-300"
  data-test="Banner"
  role="alert"
>
  <aside
    class="mr-2 ml-1"
  />
  <div
    class="flex-1"
  />
</div>
`;
